core/lib/zksync_core/src/sync_layer/gossip/conversions.rs

```diff
@@ -0,0 +1,88 @@
+//! Conversion logic between server and consensus types.
+
+use anyhow::Context as _;
+use serde::{Deserialize, Serialize};
+
+use zksync_consensus_roles::validator::{
+    BlockHeader, BlockHeaderHash, BlockNumber, CommitQC, FinalBlock, Payload,
+};
+use zksync_types::block::ConsensusBlockFields;
+use zksync_types::{
+    api::en::SyncBlock, Address, L1BatchNumber, MiniblockNumber, ProtocolVersionId, H256,
+};
+
+use crate::sync_layer::fetcher::FetchedBlock;
+
+// FIXME: should use Protobuf
+#[derive(Debug, Serialize, Deserialize)]
```

Comment:
IMO, using JSON serialization is fine until the logic is actually integrated into EN.